### ISSUE TITLE
Improve Rails support

### ISSUE THREAD

**Issue by mschwager** [mschwager/route-detect#8]
Ruby Rails support for route detection and authn/authz detection is currently limited. This is primarily due to two factors:

- [Convention over configuration](https://stackoverflow.com/questions/71985512/convention-over-configuration-in-rails). This means Rails' connection between a route and its controller is implicitly defined. And authn/authz properties are often defined in the controller, so connecting that information to a route requires connecting the controller. This makes it challenging for `route-detect` to connect a route with its authn/authz information.
- The implicit nature of [route definitions](https://guides.rubyonrails.org/routing.html). For example, "A single call to [resources](https://api.rubyonrails.org/v7.0.5.1/classes/ActionDispatch/Routing/Mapper/Resources.html#method-i-resources) can declare all of the necessary routes for your index, show, new, edit, create, update, and destroy actions." This is exacerbated by nested definitions, `only:` filtering, `concern:`, `shallow:`, etc.

In short, there's a lot of automagic functionality going on behind the scenes in Rails routing that makes statically analyzing it via Semgrep rules challenging. This issue exists to document this shortcoming and brainstorm possibilities for improvement.
